src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferReader.cs

```diff
@@ -94,36 +102,70 @@ private void MoveNext()
             _end = true;
         }
 
-        public void Skip(int length)
+        public void Skip(int byteCount)
         {
-            if (length < 0)
+            if (byteCount < 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
 
-            _consumedBytes += length;
+            _consumedBytes += byteCount;
 
-            while (!_end && length > 0)
+            while (!_end && byteCount > 0)
             {
-                if ((_index + length) < _currentSpan.Length)
+                if ((_index + byteCount) < _currentSpan.Length)
                 {
-                    _index += length;
-                    length = 0;
+                    _index += byteCount;
+                    byteCount = 0;
                     break;
                 }
 
                 var remaining = (_currentSpan.Length - _index);
 
                 _index += remaining;
-                length -= remaining;
+                byteCount -= remaining;
 
                 MoveNext();
             }
 
-            if (length > 0)
+            if (byteCount > 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
         }
+
+        internal static int CopyTo(BufferReader<TSequence> bytes, Span<byte> destination)
```

Comment:
It doesn't follow usual `CopyTo` conventions about destination sizes. Maybe we should name it differently? `Read(span)`?

Reply:
It does not advance the index, so it cannot be called Read, but I will rename it to Peek

Reply:
Maybe it should advance? Wouldn't it be the most common use case?

Reply:
Or `TryRead(Span)`.